src/components/radio-button/radio-button.test.tsx

```diff
@@ -0,0 +1,224 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import RadioButton from ".";
+
+test("renders with provided `label`", () => {
+  render(<RadioButton value="radio1" label="Radio Button 1" />);
+
+  const radioButton = screen.getByRole("radio", { name: "Radio Button 1" });
+
+  expect(radioButton).toBeInTheDocument();
+  expect(screen.getByTestId("radio-svg")).toBeVisible();
+  expect(screen.getByText("Radio Button 1")).toBeVisible();
+});
+
+test("calls `onClick` when provided and the radio button is clicked", async () => {
+  const user = userEvent.setup();
+  const onClick = jest.fn();
+  render(
+    <RadioButton value="radio1" label="Radio Button 1" onClick={onClick} />
+  );
+
+  const radioButton = screen.getByRole("radio");
+
+  await user.click(radioButton);
+
+  expect(onClick).toHaveBeenCalledTimes(1);
+});
+
+test("calls `onChange` when provided and the radio button is clicked", async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(
+    <RadioButton value="radio1" label="Radio Button 1" onChange={onChange} />
+  );
+
+  const radioButton = screen.getByRole("radio");
+
+  await user.click(radioButton);
+
+  expect(onChange).toHaveBeenCalledTimes(1);
+});
+
+test("calls `onFocus` when provided and the radio button is focused", async () => {
+  const user = userEvent.setup();
+  const onFocus = jest.fn();
+  render(
+    <RadioButton value="radio1" label="Radio Button 1" onFocus={onFocus} />
+  );
+
+  await user.tab();
```

Comment:
**suggestion** (non-blocking): I'd prefer in these focus/blur tests to explicitly fetch a particular element to call `.focus()` (and then tab/click as necessary for the rest of the test) - this ensures we're focusing/blurring the correct thing and avoids any future test flakiness if we later introduce other tab stops etc

Reply:
The focus method seems to call the actual HTML focus() method but this isn't getting flagged by the linter (issue raised [here](https://github.com/testing-library/eslint-plugin-testing-library/issues/752)). I do agree that using tab() might be a bit flakey, though I think using click() should be fine.

Reply:
hmm, I wasn't aware there was any recommendation against using `.focus()` on an HTML element, I thought this was what you needed to do to ensure a particular element was focused at the start of a test (I agree that once into the actual sequence of steps you're testing you should do "user things" like click/tab). I've done it myself in plenty of RTL tests recently and never been suggested not to 😂